### PR TITLE
[setup] Allow newer version of azure cli and ray

### PR DESCRIPTION
**Pull request by Michaelvll**
Describe the changes in this PR:
This is to update the azure cli and ray version to avoid strict version pinning.

Tested (run the relevant ones):

- [x] Any manual or new tests for this the PR
  - Create a docker container and install skypilot; `az login`; `sky launch -c azure-test --cloud azure`
- [x] `bash tests/run_smoke_tests.sh` in a new docker container

### CODE REVIEW

File: sky/backends/backend_utils.py [hunk 1]
```diff
@@ -2079,6 +2079,7 @@ def _refresh_cluster(cluster_name):
                 force_refresh=True,
                 acquire_per_cluster_status_lock=True)
         except (exceptions.ClusterStatusFetchingError,
+                exceptions.CloudUserIdentityError,
```
Comment (romilbhardwaj) [inline sky/backends/backend_utils.py:2082]: Good to catch this here! Just checking  - is this related to azure version bump?

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll! Just tried it out with azure-cli 2.43.0 and ray 2.2.0 - seems to work nicely.

File: sky/backends/backend_utils.py (hunk 1 above)
Comment (Michaelvll) [reply sky/backends/backend_utils.py:2082]: No. It is not directly related, but it is to avoid the problem where the whole `sky status -r` fails due to one of the azure cluster failed to get the user identity (not needed).
